fs/vfs/fs_read.c

```diff
@@ -83,16 +84,14 @@ ssize_t file_read(FAR struct file *filep, FAR void *buf, size_t nbytes)
    * method?
    */
 
-  else if (inode != NULL && inode->u.i_ops && inode->u.i_ops->read)
+  else if (inode != NULL && inode->u.i_ops && inode->u.i_ops->readv)
     {
       /* Yes.. then let it perform the read.  NOTE that for the case of the
        * mountpoint, we depend on the read methods being identical in
        * signature and position in the operations vtable.
        */
 
-      ret = (int)inode->u.i_ops->read(filep,
-                                     (FAR char *)buf,
-                                     (size_t)nbytes);
+      ret = inode->u.i_ops->readv(filep, iov, iovcnt);
     }
```

Comment:
let's fallback to compact_readv here if readv is NULL, instead changing each individual driver or fs.

Reply:
done